File: @LV1/1.9/egypt/egypt.hs
import Control.Monad
import Data.List

isvalid :: (Num a, Ord a) => [a] -> Bool
isvalid [a,b,c] = a ^ 2 + b ^ 2 == c ^ 2

main :: IO ()
main = do
  l <- sort . map read . words <$> getLine
  unless (all (==0) l) $ do
    if isvalid l then putStrLn "right" else putStrLn "wrong"
    main

File: LV1/1.9/egypt/egypt.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	for {
		l := make([]int, 3)
		fmt.Scanf("%d %d %d", &l[0], &l[1], &l[2])

		if l[0] == 0 && l[1] == 0 && l[2] == 0 {
			break
		}

		sort.Ints(l)

		if l[0]*l[0]+l[1]*l[1] == l[2]*l[2] {
			fmt.Println("right")
		} else {
			fmt.Println("wrong")
		}
	}
}
